Adds getting started docs with React Native Frameworks (#4086)
facebook/react-native-website: docs/getting-started.md

---
id: environment-setup
title: Get Started with React Native
hide_table_of_contents: true
---

import PlatformSupport from '@site/src/theme/PlatformSupport';
import BoxLink from '@site/src/theme/BoxLink';

> **React Native allows developers who know React to create native apps. At the same time, native developers can use React Native to gain parity between native platforms by writing common features once.**

Reviewer: Too much bold here, would not make for better readability

> We believe that the best way to experience React Native is through a Framework, a toolbox with all the necessaries APIs to let you build production ready apps.

Reviewer: ```suggestion
We believe that the best way to experience React Native is through a **Framework**, a toolbox with all the necessaries APIs to let you build production ready apps.
```

> We believe that the best way to experience React Native is through a **Framework**, a toolbox with all the necessaries APIs to let you build production ready apps.

Reviewer: ```suggestion
We believe that the best way to experience React Native is through a **Framework**, a toolbox with all the necessary APIs to let you build production ready apps.
```

> You can also use React Native without a Framework, however we’ve found that most developers benefit from using a React Native Framework like Expo. Expo provides features like file-based routing, high-quality universal libraries, and the ability to write plugins that modify native code without having to manage native files.

Reviewer: link to Expo webpage?

<details>

> <summary>Can I use React Native without a Framework?</summary>

Reviewer 2: ```
You can also use React Native without a Framework ...
```

shortly after...

```
<summary>Can I use React Native without a Framework?</summary>
```

it seems like the paragraph beginning with "You can also.." says a lot of the same things as the details/summary section

Author: Perhaps we could remove "You can also use React Native without a Framework, however " in the next paragraph, and so it would read like: 

```
We believe that the best way to experience React Native is through a **Framework**, a toolbox with all the necessaries APIs to let you build production ready apps.

We’ve found that most developers benefit from using a React Native Framework like [Expo](https://expo.dev). Expo provides features like file-based routing, high-quality universal libraries, and the ability to write plugins that modify native code without having to manage native files.
```

Reviewer 3: Starting with "Yes" might lend a little clarity, for instance, "Yes, but we believe the best way..."

In short, you’ll be able to spend time writing your app instead of writing an entire Framework yourself in addition to your app.

The React Native community has spent years refining approaches to navigation, accessing native APIs, dealing with native dependencies, and more. Most apps need these core features. A React Native Framework provides them from the start of your app.

Without a Framework, you’ll either have to write your own solutions to implement core features, or you’ll have to piece together a collection of pre-existing libraries to create a skeleton of a Framework. This takes real work, both when starting your app, then later when maintaining it.

If your app has unusual constraints that are not served well by a Framework, or you prefer to solve these problems yourself, you can make a React Native app without a Framework using Android Studio, Xcode. If you’re interested in this path, learn how to [set up your environment](set-up-your-environment) and how to [get started without a framework](get-started-without-a-framework).

</details>

## Start a new React Native project with Expo

> <PlatformSupport platforms={['android', 'ios', 'tv', 'web']} />

Reviewer: is MacOS also supported here?

Author: Great question! I don't think so, but I may be wrong. @brentvatne, can you follow up here?

Reviewer 2: it is supported, but not fully (expo modules core and a few modules support - which is good enough for usage in orbit). the main missing piece is expo cli support, we'll probably follow up with that in the next cycle. unsure how we should communicate partial support here

Author: Let's wait and add a platform tag for macOS once we support it more fully.

Expo is a production-grade React Native Framework. Expo provides developer tooling that makes developing apps easier, such as file-based routing, a standard library of native modules, and much more.

> Expo's Framework is open source and free, with an active community on [GitHub](https://github.com/expo) and [Discord](https://chat.expo.dev).

Reviewer: ```suggestion
Expo's Framework is **open source** and **free to use**, with an active community on [GitHub](https://github.com/expo) and [Discord](https://chat.expo.dev).
```

Reviewer 3: "free and open source" is the most common phrasing IME (like "FOSS")

> The team behind Expo also provides Expo Application Services (EAS), an optional set of services that complements the Framework in each step of the development process.

Reviewer: I think we should callout that the Expo team is closely collaborating with the React Native team to bring the latest features available inside Expo SDK

> 

Reviewer: Should we mention somewhere Expo vs EAS?

Author: I updated this to say:

"The team at Expo also provides Expo Application Services (EAS), an optional set of services that complements **Expo, the Framework,** in each step of the development process."

Bolded the changed part.

Happy to make any other adjustments here if this is not contrast-y enough.

To create a new Expo project, run the following in your terminal:

```shell
npx create-expo-app@latest
```

Once you’ve created your app, check out the rest of Expo’s getting started guide to start developing your app.

> <BoxLink href="https://docs.expo.dev/get-started/expo-go">Continue with Expo</BoxLink>

Reviewer 4: ```suggestion
<BoxLink href="https://docs.expo.dev/get-started/set-up-your-environment">Continue with Expo</BoxLink>
```

Nit, correct link to avoid redirect.
